buzsaki_lab_to_nwb/grosmark_code/grosmarkbehaviordatainterface.py

```diff
@@ -0,0 +1,97 @@
+"""Authors: Cody Baker and Ben Dichter."""
+from nwb_conversion_tools.basedatainterface import BaseDataInterface
+from pynwb import NWBFile
+from pynwb.file import TimeIntervals
+from pynwb.behavior import SpatialSeries, Position
+from hdmf.backends.hdf5.h5_utils import H5DataIO
+import os
+import numpy as np
+from pathlib import Path
+from scipy.io import loadmat
+from ..neuroscope import get_events, check_module
+
+
+class GrosmarkBehaviorInterface(BaseDataInterface):
+    """Primary data interface for behavioral aspects of the GrosmarkAD dataset."""
+
+    @classmethod
+    def get_input_schema(cls):
+        """Return subset of json schema for informing the NWBConverter of expepcted input arguments."""
+        return dict(properties=dict(folder_path="string"))
+
+    def convert_data(self, nwbfile: NWBFile, metadata_dict: dict,
+                     stub_test: bool = False, include_spike_waveforms: bool = False):
+        """Convert the behavioral portion of a particular session of the GrosmarkAD dataset."""
+        session_path = self.input_args['folder_path']
+        subject_path, session_id = os.path.split(session_path)
+
+        # Stimuli
+        [nwbfile.add_stimulus(x) for x in get_events(session_path)]
+
+        # States
+        sleep_state_fpath = os.path.join(session_path, "{session_id}.SleepState.states.mat")
+        # label renaming specific to Watson
+        state_label_names = dict(WAKEstate="Awake", NREMstate="Non-REM", REMstate="REM")
+        if os.path.isfile(sleep_state_fpath):
+            matin = loadmat(sleep_state_fpath)['SleepState']['ints'][0][0]
+
+            table = TimeIntervals(name='states', description="Sleep states of animal.")
+            table.add_column(name='label', description="Sleep state.")
+
+            data = []
+            for name in matin.dtype.names:
+                for row in matin[name][0][0]:
+                    data.append(dict(start_time=row[0], stop_time=row[1], label=state_label_names[name]))
+            [table.add_row(**row) for row in sorted(data, key=lambda x: x['start_time'])]
+            check_module(nwbfile, 'behavior', 'contains behavioral data').add_data_interface(table)
+
+        # Position
+        pos_filepath = Path(session_path) / f"{session_id}.position.behavior.mat"
+        pos_mat = loadmat(str(pos_filepath.absolute()))
+        starting_time = float(pos_mat['position']['timestamps'][0][0][0])  # confirmed to be a regularly sampled series
+        rate = float(pos_mat['position']['timestamps'][0][0][1]) - starting_time
+        if pos_mat['position']['units'][0][0][0] == 'm':
+            conversion = 1.0
+        else:
+            conversion = np.nan
```

Comment:
maybe we could print a warning here

Reply:
```
warnings.warn(f"Spatial units ({pos_mat['position']['units'][0][0][0]}) not listed in meters; "
                          "setting conversion to nan.")
            conversion = np.nan
```
How does that look? Do we have a go to package for more generally converting the scale units based on string text like 'cm', etc.?